admin/bin/dump-browser-stats.js

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env node
+'use strict'
+
+require('dotenv').config()
+const winston = require('winston')
+const csv = require('fast-csv')
+const chalk = require('chalk')
+const fs = require('fs')
+winston.level = 'info'
+const moment = require('moment')
+const useragent = require('useragent')
+const R = require('ramda')
+
+const dateService = require('../services/date.service')
+const poolService = require('../services/data-access/sql.pool.service')
+const completedCheckDataService = require('../services/data-access/completed-check.data.service')
+const psUtilService = require('../services/psychometrician-util.service')
+
+const outputFilename = 'browser-stats.csv'
+
+function getCheckDate (check) {
+  const checkStart = psUtilService.getClientTimestampFromAuditEvent('CheckStarted', check)
+  let checkDate = ''
+
+  if (checkStart && checkStart !== 'error') {
+    const checkStartDate = moment(checkStart)
+    checkDate = checkStartDate.isValid() ? dateService.formatUKDate(checkStartDate) : ''
+  }
+  return checkDate
+}
+
+function getBrowserStats (check) {
+  const agent = useragent.lookup(R.path(['data', 'device', 'navigator', 'userAgent'], check))
```

Comment:
in case `userAgent` is not found within the path, is that working without issues?

Reply:
`useragent` has sensible defaults.

```
> node -i
> const userAgent = require('useragent')
undefined
> userAgent.lookup(undefined)
Agent { family: 'Other', major: '0', minor: '0', patch: '0', source: '' }
```